clang/lib/Driver/ToolChains/MSVC.cpp

```diff
@@ -142,6 +142,9 @@ void visualstudio::Linker::ConstructJob(Compilation &C, const JobAction &JA,
     CmdArgs.push_back("-defaultlib:sycl-devicelib-host.lib");
   }
 
+  if (C.getDriver().IsCLMode() && Args.hasArg(options::OPT_fsycl))
+    CmdArgs.push_back("-defaultlib::sycl-devicelib-host.lib");
+
```

Comment:
Changes are good, can we add a test to check as well?

Reply:
Hi, @mdtoguchi 
Test is added, could you review again?
Thanks very much.

Reply:
Is there any particular reason why this lib is being added to the `link` command line and not being added to the default directives of the generated host object via `--dependent-lib=sycl-device-lib-host`?

Reply:
Hi, @mdtoguchi 
Just checked the code, “dependent-lib=sycl-devicelib-host” is the right approach, I will update the patch.

Thanks very much.